<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>archive_b | popucui</title>
  <meta name="author" content="Jack">
  
  <meta name="description" content="Bioinformatics | Computer">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="popucui"/>

  
    <meta property="og:image" content="undefined"/>
  

  <link href="/favicon.png" rel="icon">
  <link rel="alternate" href="/atom.xml" title="popucui" type="application/atom+xml">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->
  <script src="http://cdn.staticfile.org/jquery-i18n/1.1.1/jquery.i18n.min.js"></script>
  
</head>

<body>
  <header id="header" class="inner"><div class="alignleft">
  <h1><a href="/">popucui</a></h1>
  <h2><a href="/">Stuff Related to Bioinformatics Computer and More</a></h2>
</div>
<nav id="main-nav">
  <ul>
    
      <li><a href="/">Home</a></li>
    
      <li><a href="/archives">Archives</a></li>
    
      <li><a href="/about">About</a></li>
    
      <li><a href="/links">Links</a></li>
    
      <li><a href="/books">Books</a></li>
    
  </ul>
  <div class="clearfix"></div>
</nav>
<div class="clearfix"></div>
<div class="alignleft" style="margin-top: 15px">


  <iframe src="http://ghbtns.com/github-btn.html?user=popucui&type=follow"
  allowtransparency="true" frameborder="0" scrolling="0" width="132" height="20"></iframe>

</div>
<div class="clearfix"></div>
</header>
  <div id="content" class="inner">
    <div id="wrapper">
<h2 class="archive-title">2016</h2>


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/06/25/SGE新增用户配置/">SGE新增用户配置</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-06-25T08:20:47.000Z">Jun 25 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>日常的服务器管理中，经常需要添加或删除用户，对于linux用户的添加比较简单，使用<code>useradd</code>就可以：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#&#22686;&#21152;&#19968;&#20010;&#21517;&#20026;&#34;zhangshan&#34;&#30340;&#29992;&#25143;&#65292;&#24182;&#25226;&#20182;&#21152;&#20837;&#34;group001&#34;&#32676;&#32452;(&#27880;&#65306; &#36825;&#37324;&#25152;&#26377;&#30340;&#25805;&#20316;&#65292;&#37117;&#26159;&#20197;root&#29992;&#25143;&#36523;&#20221;&#36827;&#34892;&#30340;)&#10;useradd zhangsan group001&#10;#&#26356;&#25913;&#23494;&#30721;&#10;su zhangshan&#10;passwd &#60;p@ssw0rd&#62;</span><br></pre></td></tr></table></figure>
<p>添加完之后，新的用户就可以正常使用终端ssh登录集群（服务器）了，如果集群配置了SGE之类的作业调度系统，通常还需要对这些用户开通相应的集群队列权限，还是以用户”zhangshan”为例，将他加入”arusers”这个userset（SGE 称为Access List），这样他就可以使用”arusers”可调度的集群队列（all.q）了。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#&#23558;&#34;zhangshan&#34; &#28155;&#21152;&#21040;&#34;arusers&#34;&#29992;&#25143;&#32452;&#10;[root@ngnode1 cuijie]# qconf -au zhangshan arusers&#10;added &#34;zhangshan&#34; to access list &#34;arusers&#34;&#10;#&#26174;&#31034;&#34;arusers&#34;&#29992;&#25143;&#21015;&#34920;&#65292;&#30475;&#21040;&#34;zhangshan&#34;&#24050;&#32463;&#21152;&#20837;&#36825;&#20010;&#21015;&#34920;&#10;[root@ngnode1 cuijie]# qconf -su arusers&#10;name    arusers&#10;type    ACL&#10;fshare  0&#10;oticket 0&#10;entries chengchaoze,cuijie,zhouyi,leijie,zhangshan&#10;#&#25226;&#34;zhangshan&#34;&#20174;&#34;arusers&#34;&#21024;&#38500;&#10;[root@ngnode1 cuijie]# qconf -du zhangshan arusers&#10;deleted user &#34;zhangshan&#34; from access list &#34;arusers&#34;</span><br></pre></td></tr></table></figure>
<p>更多选项和参数，可参考<code>qconf -help</code></p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/06/05/setup-and-run-PIPS-to-identify-Putative-Pathogenicity-Islands/">setup and run PIPS to identify Putative Pathogenicity Islands</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-06-04T22:16:21.000Z">Jun 5 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>Recently I need to do PAI search using <a href="http://www.genoma.ufpa.br/lgcm/pips/" target="_blank" rel="external">PIPS</a>, after going throught all the little traps, I think it worth to write the whole process down.</p>
<ol>
<li>follow the README instructions to install hammer and tRNA-scan. Maybe you want to change the default prefix of INSTALL derectory</li>
<li>Theoretically, it should be good enough to run, but you’ll definitely be stuck at some point. OK, here are the tricks to try below</li>
<li>replace <code>gbk2embl.pl</code> with a new one, such as this:</li>
</ol>
<pre><code><span class="comment">#!/usr/bin/env perl</span>
<span class="keyword">use</span> <span class="title">strict</span>;
<span class="keyword">use</span> <span class="title">warnings</span>;
<span class="keyword">use</span> <span class="title">Bio</span>::<span class="title">SeqIO</span>;

<span class="keyword">if</span> (@ARGV != <span class="number">2</span>) {    <span class="keyword">die</span> <span class="string">"USAGE: gb2embl.pl &lt;in.gbk&gt; &lt;out.embl&gt;\n"</span>; }

my <span class="variable">$seqio</span> = Bio::SeqIO-&gt;new(<span class="string">'-format'</span> =&gt; <span class="string">'genbank'</span>, <span class="string">'-file'</span> =&gt; <span class="string">"$ARGV[0]"</span>);
my <span class="variable">$seqout</span> = <span class="keyword">new</span> Bio::SeqIO(<span class="string">'-format'</span> =&gt; <span class="string">'embl'</span>, <span class="string">'-file'</span> =&gt; <span class="string">"&gt;$ARGV[1]"</span>);
<span class="keyword">while</span>( my <span class="variable">$seq</span> = <span class="variable">$seqio</span>-&gt;next_seq) {
  <span class="variable">$seqout</span>-&gt;write_seq(<span class="variable">$seq</span>)
}
</code></pre><p>For the last step, modify <code>mergfiles.pl</code>. change for e.g. <code>%hash2-&gt;{$chave}</code> to <code>$hash2{$chave}</code>, note that a couple of similar syntax mistake in this script.</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/05/29/Fast-genome-and-metagenome-distance-estimation-using-MinHash/">Fast genome and metagenome distance estimation using MinHash</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-05-29T06:47:55.000Z">May 29 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p><a href="http://mash.readthedocs.org" target="_blank" rel="external">mash</a> 是一个可快速计算两个（或多个）基因组之间相似度的软件，可输入基因组序列或测序结果（fastq文件），使用计算机科学中一个经典的方法MinHash，一般用来判断两个网页内容是否雷同，以决定是否从搜索结果中去掉某些类似的结果。</p>
<p>使用示例：</p>
<p><code>mash dist -p 4 -l 、/path/to/ref.fa query_fa.txt &gt; query_mashed.txt</code></p>
<p>用于计算所有query序列和ref序列的相似度，而相似度和物种间的亲缘关系是相关的，其中<query_fa.txt> 是一个记录了所有query序列的文件</query_fa.txt></p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/02/03/parsing-ABI-file/">parsing ABI file</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-02-03T14:45:00.000Z">Feb 3 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>ABI测序仪下机的结果包含.ab1和.seq文件，通常可以导入BioEdit这样的图形化软件检查峰形和运行blast，而如果想要实现自动化的检测杂合位点并做比对，最终得到变异位点，就需要花些时间研究下。</p>
<p>主要的问题是，ab1文件是二进制的，关于文件格式的文档稀少，唯一靠谱的是<a href="http://www.appliedbiosystems.com/support/software_community/ABIF_File_Format.pdf" target="_blank" rel="external">ABI的官方文档</a> 仔细看过这个文档你会发现，看完之后作为新手的你尽管对于这个格式有所了解，知道它分为Header、Directory以及Data，以及有很多的tag。但知道这些对于想要从ab1提取trace value并进行heterogeneous base calling是远远不够的。所好的是，已经有人有过类似需求并作了尝试。</p>
<ul>
<li><p><a href="https://github.com/bow/abifpy" target="_blank" rel="external">abifpy</a> 实现的一个解析ab1文件的模块，现已并入BioPython。优点是兼容Python2和Python3，部分meta data已经给你提取出来了，而如果想要更多，可以使用get_data(‘<em>youkey</em>‘)获取。</p>
</li>
<li><p>如果只需要取出每个通道的trace value，有了官方文档和abifpy就够了，如果还想学习更多，可以读一读<a href="https://code.google.com/archive/p/seqtrace/" target="_blank" rel="external">SeqTrace</a>源码，从中可以学到怎样组织一个项目的代码，以及怎么写测试。</p>
</li>
<li><p>前面两个都研究一番之后，还是不能满足你的好奇心？可以把<a href="http://sourceforge.net/projects/staden/files/io_lib/" target="_blank" rel="external">io lib</a>这个库研究下，学习使用它的API.</p>
</li>
</ul>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  

  <nav id="pagination">
  
  
  <div class="clearfix"></div>
</nav>
</div>
    <div class="clearfix"></div>
  </div>
  <footer id="footer" class="inner"><div class="alignleft">
  
  &copy; 2016 Jack
  
</div>
<div class="clearfix"></div></footer>
  <script src="http://cdn.bootcss.com/jquery.imagesloaded/2.1.0/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>




<link rel="stylesheet" href="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.css" media="screen" type="text/css">
<script src="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>


</body>
</html>