<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Archives | popucui</title>
  <meta name="author" content="Jack">
  
  <meta name="description" content="Bioinformatics | Computer">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="popucui"/>

  
    <meta property="og:image" content="undefined"/>
  

  <link href="/favicon.png" rel="icon">
  <link rel="alternate" href="/atom.xml" title="popucui" type="application/atom+xml">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->
  <script src="http://cdn.staticfile.org/jquery-i18n/1.1.1/jquery.i18n.min.js"></script>
  
</head>

<body>
  <header id="header" class="inner"><div class="alignleft">
  <h1><a href="/">popucui</a></h1>
  <h2><a href="/">Stuff Related to Bioinformatics Computer and More</a></h2>
</div>
<nav id="main-nav">
  <ul>
    
      <li><a href="/">Home</a></li>
    
      <li><a href="/archives">Archives</a></li>
    
      <li><a href="/links">Links</a></li>
    
      <li><a href="/books">Books</a></li>
    
  </ul>
  <div class="clearfix"></div>
</nav>
<div class="clearfix"></div>
<div class="alignleft" style="margin-top: 15px">


  <iframe src="http://ghbtns.com/github-btn.html?user=popucui&type=follow"
  allowtransparency="true" frameborder="0" scrolling="0" width="132" height="20"></iframe>

</div>
<div class="clearfix"></div>
</header>
  <div id="content" class="inner">
    <div id="wrapper">
<h2 class="archive-title">Archives</h2>


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2015/09/18/hello-again/">hello again</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2015-09-17T23:10:13.000Z">Sep 18 2015</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>翻了一下以前的文章发现，最近的上一篇文章是2013年11月的，那时候还在南京的学校，从找工作开始，然后面试，来北京，上班之后就更没有留出时间静下来写东西了。后来换了Mac，还特意把blog的资料作了备份，想着不久之后，重搭环境另起炉灶的，没想一下子拖到不久前才在新电脑上配好环境。</p>
<p>上班的这一年多，改变了很多。工作上从项目执行到流程负责人，再到信息分析主管而后技术副总监，见证了公司规模的扩张和变迁，也目睹了身边同事的来来去去，“送走”了三波运营岗同事，现在接触的大都是第四波的人。公司在成长，自己的工作也在经历变化，以前只需要自己干活，现在要考量人和事，把相应的事交给合适的人去做，然后跟踪进度，培养新人。</p>
<p>除了工作，生活上也有一些变化。刚到北京住的没有独立卫生间的小院，后来的院子有单独卫生间，尽管很少，还是方便多了。随公司搬到天津之后，住的更好。经过两段感情经历的洗礼，也成长了很多，对自己也有了更多地认识。写东西是一个特别好的总结的方式，通过它可以和自己对话，从而自省；和别人交流，从而结交同好。所以，以后我会坚持下去的。</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/11/24/retrieving-and-sending-infomation-over-the-web/">retrieving and sending infomation over the web</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-11-24T08:04:53.000Z">Nov 24 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>I found <a href="http://pythonforbiologists.com/index.php/applied-python-for-biologists/applied-python-2/" target="_blank" rel="external">this</a> when hanging around online. To be honest, I am dying to learn web crawling long time ago, can not wait to play with this.</p>
<p>modules involved:</p>
<ul>
<li>urllib2</li>
<li>re</li>
<li><a href="http://www.crummy.com/software/BeautifulSoup/" target="_blank" rel="external">BeautifulSoup</a> Need to install manually</li>
</ul>
<p>This post is very easy to follow with basic familarity of Python, the difficulty of web crawling is lots of, if not all, websites follow HTML specification exactly, there are often errors here and there if you try to parse useful bits of infomation out there, and here comes BeautifulSoup which handle most of these for us.</p>
<p>The second section discuss sending email with Python. One thing need to keep in mind is check your internet connection first if your code refuse to work :-)- I have complete the spec of No.5, and below are some commands I find useful:</p>
<ul>
<li><code>execfile(&#39;script.py&#39;)</code> #load script into console</li>
<li><code>python -i script.py</code> #execute script then come to Python console, ‘i’ short for ‘interactive’</li>
<li>sometimes you need to insert a time stamp into you file, and <code>time.ctime()</code> is probably the simplest solution<br>source code is <a href="https://github.com/popucui/RetrievandMail/blob/master/RetrievandMail.py" target="_blank" rel="external">here</a></li>
</ul>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/10/27/一道信息分析面试题的分析/">一道信息分析面试题的分析</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-10-27T06:47:39.000Z">Oct 27 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>前几天接到一个公司的电话面试，当天晚上HR发来一道题，限三日内完成并发回。我看了下，正好用到的知识自己都熟悉，花了几个钟头就给发回去了 :-)</p>
<p>原题大意这样 写一个程序，分别计算一个目录里所有fasta格式的核酸序列的GC含量，然后按照GC含量由低到高显示，并把结果写入一个文件<br>显示序列信息的格式为：<br>文件名  序列名  GC含量<br>hem.fasta hem1  24.3%</p>
<p>这道题挺简单的，细看之下主要有这么几点：</p>
<ul>
<li>计算序列GC含量，可以用一个函数实现</li>
<li>列出目录下的所有文件，然后判断他们是不是fasta格式文件。前者用os模块的<code>listdir</code>就可，后者就是要看文件的首行是不是形如“&gt;at1g23490”开头，即是否“&gt;”开头，后面紧跟字符，正则就行了。</li>
<li>结果排序。这个应该是这道简单笔试题最难的部分了。正好之前从一个<a href="http://wangwei007.blog.51cto.com/68019/1131610/" target="_blank" rel="external">博客</a>读到解决排序的办法，就依样做了。把每条序列的结果存到一个列表里，然后对列表排序。</li>
</ul>
<p>这是我的代码 <a href="https://github.com/popucui/PyManipulateFiles/blob/master/bioinfoInterviw.py" target="_blank" rel="external">信息分析面试</a></p>
<p> EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/10/11/linux常用命令/">linux常用命令</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-10-11T15:13:13.000Z">Oct 11 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <h3 id="git">git</h3><p>在github上参与别人的项目或则多人共同开发一个项目的话，fork一个项目的仓库过来，自己做完提交pull request之后，就等着作者review过之后进行merge了。但是之后作者再进行新的改动，我的分支上还是原样，那么怎样让把我的分支上的与最新的仓库一直呢?</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#assume you originally cloned from your forked repo,that remote will be called &#34;origin&#34;, &#10;#then you need to add firstguy repo as another remote&#10;git remote add firstguy git://github.com/firstguy/repo.git&#10;#After that is all set up, you should indeed be able to&#10;git pull firstguy mastergit push origin</span><br></pre></td></tr></table></figure>
<h3 id="ubuntu_apt">ubuntu apt</h3><p>apt search</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"># package search&#10;apt-cache search &#10;apt-cache show #------(package &#33719;&#21462;&#21253;&#30340;&#30456;&#20851;&#20449;&#24687;&#65292;&#22914;&#35828;&#26126;&#12289;&#22823;&#23567;&#12289;&#29256;&#26412;&#31561;)</span><br></pre></td></tr></table></figure>
<p>apt install&amp;remove</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#package install&#10;sudo apt-get install &#10;sudo apt-get install # -----(package - - reinstall &#37325;&#26032;&#23433;&#35013;&#21253;)&#10;sudo apt-get -f install # -----(&#24378;&#21046;&#23433;&#35013;?#&#34;-f = --fix-missing&#34;&#24403;&#26159;&#20462;&#22797;&#23433;&#35013;&#21543;...)&#10;sudo apt-get remove #-----(package &#21024;&#38500;&#21253;)&#10;sudo apt-get remove - - purge # ------(package &#21024;&#38500;&#21253;&#65292;&#21253;&#25324;&#21024;&#38500;&#37197;&#32622;&#25991;&#20214;&#31561;)&#10;sudo apt-get autoremove --purge # ----(package &#21024;&#38500;&#21253;&#21450;&#20854;&#20381;&#36182;&#30340;&#36719;&#20214;&#21253;+&#37197;&#32622;&#25991;&#20214;&#31561;</span><br></pre></td></tr></table></figure>
<p>apt update and more</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#update source&#10;sudo apt-get update &#10;sudo apt-get upgrade #------&#26356;&#26032;&#24050;&#23433;&#35013;&#30340;&#21253;&#10;sudo apt-get dist-upgrade # ---------&#21319;&#32423;&#31995;&#32479;&#10;sudo apt-get dselect-upgrade #------&#20351;&#29992; dselect &#21319;&#32423;&#10;apt-cache depends #-------(package &#20102;&#35299;&#20351;&#29992;&#20381;&#36182;)&#10;apt-cache rdepends # ------(package &#20102;&#35299;&#26576;&#20010;&#20855;&#20307;&#30340;&#20381;&#36182;?#&#24403;&#26159;&#26597;&#30475;&#35813;&#21253;&#34987;&#21738;&#20123;&#21253;&#20381;&#36182;&#21543;...)&#10;sudo apt-get build-dep # ------(package &#23433;&#35013;&#30456;&#20851;&#30340;&#32534;&#35793;&#29615;&#22659;)&#10;apt-get source #------(package &#19979;&#36733;&#35813;&#21253;&#30340;&#28304;&#20195;&#30721;)&#10;sudo apt-get clean &#38;&#38; sudo apt-get autoclean # --------&#28165;&#29702;&#19979;&#36733;&#25991;&#20214;&#30340;&#23384;&#26723; &#38;&#38; &#21482;&#28165;&#29702;&#36807;&#26102;&#30340;&#21253;&#10;sudo apt-get check #-------&#26816;&#26597;&#26159;&#21542;&#26377;&#25439;&#22351;&#30340;&#20381;&#36182;</span><br></pre></td></tr></table></figure>
<p><strong>nohup</strong>: short for “no hang up”</p>
<ul>
<li>usage: nohup command &amp;</li>
<li>default output: nohup.out which contains std out and std error, I refquently use this: <code>nohup sslocal &gt; log &amp;</code></li>
</ul>
<p>EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/10/11/no-desktop-when-installing-windows7-solved/">no desktop when installing windows7 solved</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-10-11T14:44:50.000Z">Oct 11 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p><strong>Here is the issue</strong>: When I am trying to help someone to reinstall wondows7 on a TOSHIBA laptop, I can not get into the GUI interface, no matter with windows PE, USB disk created by universal USB  installer or windows 7 DVD download tool which is released by microsoft.</p>
<p>After all these failed, I turn to ubuntu. I plug in my USB disk with ubuntu 12.04 inside, press F12 when booting, and I finally can have my desktop where I can browe all files on the hard drive. I had to erase the system partition when the attempt to check and repair the possible error in system partition failed me.</p>
<p>Then all things go as expected, so you see, that is one reason to learn linux :-) -</p>
<p>EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/23/differential-expression-for-rna-seq-data/">differential expression for RNA-Seq data</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-23T12:24:45.000Z">Sep 23 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>从<a href="http://www.rna-seqblog.com/" target="_blank" rel="external">rna-seqblog</a>看到<a href="http://genomebiology.com/2013/14/9/R95/abstract" target="_blank" rel="external">这篇paper</a>,他们对流行的寻找差异基因的方法多了评测，压根没提到DEGSeq,我看到在bioconductor上面，DEGSeq的下载量只有DESeq、edgeR的一半，所以人家说‘流行的’就把DEGSeq排除在外了。</p>
<p>不过这里要说的主角是另一本书，在翻看前述论文时看到的，打来不禁有相见恨晚的感叹，<a href="http://link.springer.com/book/10.1007/978-1-4614-0782-9/page/1" target="_blank" rel="external">原书</a>还没看完，这里把里面的一些要点记录下来。</p>
<h3 id="microarray和sequencing">microarray和sequencing</h3><p>RNA-Seq相比microarray的优点已经被谈得够多了，这里作者同样讲了RNA-Seq的不足：</p>
<ul>
<li>RNA-Seq数据有GC偏好</li>
<li>mapping的不确定对基因组不同区域的影响是不一样的</li>
<li>RNA-Seq比较不同处理下的基因表达水平有偏好性。RNA-Seq的统计方法对高counts的检测能力要比低counts强，也就是说RNA-Seq会倾向于将长度更长的基因算作差异（DE）基因。</li>
<li>RNA-Seq的高灵敏性带来的问题是，提取、富集RNA，RNA片段化和转化成cDNA的过程中都有更大的带来偏差的可能。如小RNA提取的方法强烈影响得到的序列集合。</li>
<li>RNA-Seq数据更复杂，数据分析可能是瓶颈。</li>
</ul>
<h3 id="mapping">mapping</h3><p>主要有两类方法：基于hash-tables;基于Burrows Wheeler Transform</p>
<ul>
<li>hash-tables优点是可以找出用户指定的所有structural variants,确定是内存要求高。</li>
<li>BWT相比前者使用内存更经济，缺点是，BWT确认错配的方法涉及把每个read的大量可能组合进行比对，二这需要相当大的计算量。很多实现方法对此的处理是，不穷尽所有错配比对，这也就意味着一些正确的比对丢失了。</li>
</ul>
<h3 id="RNA-Seq实验设计">RNA-Seq实验设计</h3><ul>
<li>生物学重复重要性：如果我们比较一个癌症组织和一个正常组织，那么我们下的任何一个结论都不能推广到其癌组织群和正常组织群，因为分析过程没有考虑癌组织与癌组子和正常组织与正常组织间潜在的巨大差异。我们的结论仅适用于该研究的样本。</li>
<li>上样的时候要考虑到不同lane，不同flow cell之间存在的系统差异</li>
<li>reads长度的确定。更长的reads花费多，但map到trancriptome会更多，对外显子-外显子间连接区分能力更强，发现更多的遗传变异，但是DE的发现来自reads密度，而不是序列覆盖度。总之，是用更长的reads来换取‘mappability’，还是用更深的测序深度换取更多的reads密度呢？作者给的答案是，投资与‘mappability’所获通常比较小。</li>
<li>paried-end reads: PE-reads可以有更多的reads map到转录组或基因组上，有助于发现结构变异，然而PE-reads带来的更多花费并不能增加检测DE的强度。</li>
</ul>
<p>EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/18/generate-venn-diagram-in-r/">generate venn diagram in R</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-18T14:35:57.000Z">Sep 18 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>Our DGE project report provide a differently expressing genes list which is infered using DEGseq. When I was trying to follow the analysis pipline, I found that DEGseq has a so simple document that I cannot even figure out how to apply the function in my work, then I turn to DESeq, which provide a very good document with explanation of options and typical cases. Now I have to DG(Different Gene)lists: one from NOvogene, another is mine(generated by DESeq). Some genes are the same in them, while some not. My goal is to identify the same and plot a venn diagram. Here is the outline:</p>
<ul>
<li>idnetify the same genes, calculate the number of same and different genes.</li>
<li>draw a venn diagram in R to display above result.</li>
</ul>
<p>DEGseq list has 2k+ genes, and my DESeq list has 1230 genes, obviously I need a quick script to compare each gene in DESeq and remember the number of genes in both.<br><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">dfgnlist1 = sys.argv[<span class="number">1</span>]</span><br><span class="line"></span><br><span class="line">dfgnlist2 = sys.argv[<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="keyword">with</span> open(dfgnlist2, <span class="string">'r'</span>) <span class="keyword">as</span> deg:</span><br><span class="line"></span><br><span class="line">  dfgn2 = deg.readlines()</span><br><span class="line"></span><br><span class="line"><span class="keyword">with</span> open(dfgnlist1,<span class="string">'r'</span>) <span class="keyword">as</span> de:</span><br><span class="line"></span><br><span class="line">  dfgn1 =de.readlines()</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line"></span><br><span class="line">j = <span class="number">0</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> g <span class="keyword">in</span> dfgn1:</span><br><span class="line"></span><br><span class="line">  <span class="keyword">if</span> g <span class="keyword">in</span> dfgn2:</span><br><span class="line"></span><br><span class="line">    <span class="keyword">with</span> open(<span class="string">"samegenes.txt"</span>,<span class="string">'a'</span>) <span class="keyword">as</span> samegenes:</span><br><span class="line"></span><br><span class="line">      samegenes.write(g)</span><br><span class="line"></span><br><span class="line">      i += <span class="number">1</span></span><br><span class="line"></span><br><span class="line">  <span class="keyword">else</span>:</span><br><span class="line"></span><br><span class="line">    <span class="keyword">with</span> open(<span class="string">"dfgenes.txt"</span>,<span class="string">'a'</span>) <span class="keyword">as</span> dfgenes:</span><br><span class="line"></span><br><span class="line">      dfgenes.write(g)</span><br><span class="line"></span><br><span class="line">      j += <span class="number">1</span></span><br><span class="line"><span class="keyword">print</span> <span class="string">'========'</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> i, <span class="string">'same genes'</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> j, <span class="string">'different genes'</span></span><br></pre></td></tr></table></figure><br>
      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/14/using-htseq/">using HTSeq</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-14T00:29:50.000Z">Sep 14 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>用tophat做完mapping之后，接下来就是count reads，<a href="http://www-huber.embl.de/users/anders/HTSeq/doc/overview.html" target="_blank" rel="external">HTSeq</a>的文档很详细，还有一个use case，除了这个case，其他的全部跟着走了一遍。接下来使用htseq-qa和htseq-count的时候，遇到了问题，记录在这里：</p>
<h3 id="htseq-qa">htseq-qa</h3><p>安装了HTSeq后，htseq-qa的使用也很简单：</p>
<ul>
<li><code>htseq-qa [options] read_file</code> #assumes htseq is in your PATH</li>
<li><code>python -m HTSeq.scripts.qa [options] read_file</code> #if htseq is not in your PATH</li>
</ul>
<p>检查clean data的时候很顺利，在read_file是Col_0的raw data时，遇到了问题，原因是raw data里面有的reads quality string信息不完整，尝试过删除该reads或者把reads quality数据补全，都不成。</p>
<h3 id="htseq-count">htseq-count</h3><p>htseq-count的使用与qa类似，如果htseq不再你的PATH，可以这样：<code>python -m HTSeq.scripts.count [options] &lt;sam_file&gt; &lt;gff_file&gt;</code>。我遇到的问题有两个：</p>
<ol>
<li>tophat输出的是BAM格式的结果，而htseq-count需要SAM格式的比对结果，首先要转换，这个用samtools就行了：<code>samtools view col.bam &gt;&gt; col.sam</code> samtools默认的输入就是BAM格式，这里用重定向将屏幕输出写入col.sam</li>
<li>然后运行htseq-count报错：”ValueError: The attribute string seems to contain mismatched quotes”.搜了下，报错的行里面’gene_name’id含有多余的分号，手工修改了几次，还是报同样的错。这时候需要写一个脚本了：</li>
</ol>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> re</span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line">input_file = sys.argv[<span class="number">1</span>]</span><br><span class="line">output_file = sys.argv[<span class="number">2</span>]</span><br><span class="line">j = <span class="number">0</span></span><br><span class="line"><span class="keyword">with</span> open(input_file, <span class="string">'r'</span>) <span class="keyword">as</span> f:</span><br><span class="line">  <span class="keyword">for</span> line <span class="keyword">in</span> f.readlines():</span><br><span class="line">    gene_name = re.search(<span class="string">r'gene_name\s(\S+)'</span>,line)</span><br><span class="line">    gene_id = gene_name.group(<span class="number">1</span>)</span><br><span class="line">    match = re.search(<span class="string">r'(\w+);(\w+)'</span>, gene_id)</span><br><span class="line">    <span class="keyword">if</span> match:</span><br><span class="line">      newid = <span class="string">''</span>.join([<span class="string">'"'</span>, match.group(<span class="number">1</span>), match.group(<span class="number">2</span>), <span class="string">'"'</span>, <span class="string">';'</span>])</span><br><span class="line">      newline = line.replace(gene_id, newid)</span><br><span class="line">    <span class="keyword">else</span>:</span><br><span class="line">      newline = line</span><br><span class="line"></span><br><span class="line">    <span class="keyword">with</span> open(output_file, <span class="string">'a'</span>) <span class="keyword">as</span> newf:</span><br><span class="line">      newf.write(newline)</span><br><span class="line">      j += <span class="number">1</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> <span class="string">'totaly'</span>, j, <span class="string">'number of lines modified!'</span></span><br></pre></td></tr></table></figure>
<p>得到正确格式的gtf之后，就可以count了：<code>python -m HTSeq.scripts.count -m intersection-nonempty -s no sam_file gtf_file &gt;&gt; samplecount</code> 需要注意的事这里’-s’选项默认是’yes’,所以如果样品建库不是连特异性的话，一定要加<code>-s no</code>,要不有一半的reads就浪费了。另外就是把count结果写入文件，以备之后分析。</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/10/hello-again-ubuntu/">hello again ubuntu</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-10T10:48:24.000Z">Sep 10 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>我接触的第一个linux发行版就是基于ubuntu的Macbuntu，那时候还是10.04的时代，后来自己不小心弄坏了MBR，连Windows也进不去，捉了不少难，好在从此学会PE重建主引导的方法。之后进转投Fedora，直到现在又用回ubuntu。</p>
<p>最主要的原因就是受够了Fedora那难看的字体，而从一开始ubuntu给我的印象就是漂亮，比windows漂亮，至于Mac吗，还没用过暂不置评。放弃Fedora还有一个原因是yum包管理也开始觉得不爽，每次我要search install软件，都要先更新下，前天安装HTSeq又遇到更新源频频出问题，慢的叫人抓狂。</p>
<p>转到ubuntu之后，顺便也把博客从<a href="http://octopress.org/" target="_blank" rel="external">octopress</a>迁移到<a href="http://zespia.tw/hexo/" target="_blank" rel="external">hexo</a>。octopress是基于ruby的，而hexo基于node，速度比前者明显快多了，随着博客文章越来越多，我越来越在意这种差别。最后，把用到过得几个命令记下来：</p>
<ul>
<li><code>source somefile</code> reload the file,but only affect current session, will lose when restart terminal</li>
<li><code>dpkg -i xxx.deb</code> install xxx soft which is in ubuntu deb format</li>
<li><code>nvm use version num</code> use the specified version of node <code>nvm ls</code> list all installed versions of node</li>
</ul>
<p>EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/08/use-kobas-to-identify-enriched-pathways/">use KOBAS to identify enriched pathways</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-08T13:33:00.000Z">Sep 8 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>看着手里公司做好的KEGG富集列表，总想自己试试看自己能做怎么样。之前想着自己用python写个脚本来做，开始写才发现，这活不是一个几行脚本能搞定的，不过发现了keggrest package，一个实现了KEGG API的好东西，基本上get link list都有了。今天发现了KOBAS，才知道国人已经为傻瓜化KEGG注释搭建<a href="http://kobas.cbi.pku.edu.cn/home.do">web server</a>.我要做的就是把合适的基因号导入，然后run就行了。</p>
<h3 id="ID_prepare">ID prepare</h3><p>我的基因号现在形如：</p>
<ul>
<li>AT1G01170</li>
<li>AT1G01480</li>
<li>AT1G01490</li>
</ul>
<p>我要把它转为：</p>
<ul>
<li>ath:AT1G01170</li>
<li>ath:AT1G01480</li>
<li>ath:AT1G01490</li>
<li>ath:AT1G01610</li>
</ul>
      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  

  <nav id="pagination">
  
  
    <a href="/archives/page/2/" class="alignright next">Next</a>
  
  <div class="clearfix"></div>
</nav>
</div>
    <div class="clearfix"></div>
  </div>
  <footer id="footer" class="inner"><div class="alignleft">
  
  &copy; 2015 Jack
  
</div>
<div class="clearfix"></div></footer>
  <script src="http://cdn.bootcss.com/jquery.imagesloaded/2.1.0/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>




<link rel="stylesheet" href="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.css" media="screen" type="text/css">
<script src="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>


</body>
</html>