<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Archives | popucui</title>
  <meta name="author" content="Jack">
  
  <meta name="description" content="Bioinformatics | Computer">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="popucui"/>

  
    <meta property="og:image" content="undefined"/>
  

  <link href="/favicon.png" rel="icon">
  <link rel="alternate" href="/atom.xml" title="popucui" type="application/atom+xml">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->
  <script src="http://cdn.staticfile.org/jquery-i18n/1.1.1/jquery.i18n.min.js"></script>
  
</head>

<body>
  <header id="header" class="inner"><div class="alignleft">
  <h1><a href="/">popucui</a></h1>
  <h2><a href="/">Stuff Related to Bioinformatics Computer and More</a></h2>
</div>
<nav id="main-nav">
  <ul>
    
      <li><a href="/">Home</a></li>
    
      <li><a href="/archives">Archives</a></li>
    
      <li><a href="/about">About</a></li>
    
      <li><a href="/links">Links</a></li>
    
      <li><a href="/books">Books</a></li>
    
  </ul>
  <div class="clearfix"></div>
</nav>
<div class="clearfix"></div>
<div class="alignleft" style="margin-top: 15px">


  <iframe src="http://ghbtns.com/github-btn.html?user=popucui&type=follow"
  allowtransparency="true" frameborder="0" scrolling="0" width="132" height="20"></iframe>

</div>
<div class="clearfix"></div>
</header>
  <div id="content" class="inner">
    <div id="wrapper">
<h2 class="archive-title">Archives</h2>


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/11/30/Clinvar学习笔记/">Clinvar学习笔记</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-11-30T05:35:59.000Z">Nov 30 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <h4 id="ClinVar_database">ClinVar database</h4><p><a href="https://www.ncbi.nlm.nih.gov/clinvar/intro/" target="_blank" rel="external">ClinVar</a>是NCBI开发的一个存储人类变异和表型关系的数据库，一般会有证据支持信息（来自临床诊断还是仅仅源自文献报道）。</p>
<h5 id="数据库建立">数据库建立</h5><p>2013年发布第一个版本，初始数据由OMIM，dbSNP，GeneViewer等数据库，以及一部分临检实验室提交的数据。</p>
<h5 id="数据库结构">数据库结构</h5><p>同一个submitter不同批次提交的数据，都会被记录（前缀是SCV，不同批次间做了区分）。不同submitter提交的针对同一个位点的变异，聚合为一个RCV记录，如果他们的结论有冲突，会有一个专家组进行review后给出结论，同一个位点不一致的结论每个月会发布一个release。</p>
<p>数据收集方法</p>
<ul>
<li>临床检测：符合CLIA或ISO 1589标准的实验室报告出来</li>
<li>科学研究：科研项目鉴定到的</li>
<li>仅文献报道：被第三方引用并未声明出处，和第二点的区别是，科学研究来源的有据可查，而这种没有原文可追溯</li>
</ul>
<p>需要注意的是，<strong>GWAS</strong>的结果，除了经人工鉴别并给出临床解释的之外，其他的都没有收录。</p>
<h5 id="数据库使用">数据库使用</h5><p>数据获取方式</p>
<ul>
<li>web 可使用基因名，rs number，疾病名，HGVS表达式（人类基因组变异学会定义的用于精确描述变异信息的格式）</li>
<li>ftp XML格式存储的是全部ClinVar的信息，VCF存储的是SNP信息，大于50nt的SNV不包括在内</li>
<li>api  支持esearch efetch esummary elink共4种方法，这里是<a href="https://www.ncbi.nlm.nih.gov/clinvar/docs/maintenance_use/#api" target="_blank" rel="external">文档</a></li>
</ul>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/11/18/sqlite学习笔记/">sqlite学习笔记</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-11-18T07:18:30.000Z">Nov 18 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>最近在学习sqlite，<a href="https://www.sqlite.org/docs.html" target="_blank" rel="external">sqlite官网</a>上面的文档无疑是最权威的，但对于新手来说上手有些困难，而是比较适合已经入门的初级使用者查阅。搜索之后发现下面这两个网站，还不错，我主要是跟着第一个学习的：</p>
<ul>
<li><a href="https://www.tutorialspoint.com/sqlite/index.htm" target="_blank" rel="external">sqlite教程</a></li>
<li><a href="http://www.sqlitetutorial.net/tryit/" target="_blank" rel="external">在线版sqlite</a></li>
</ul>
<h5 id="简介">简介</h5><p>获取帮助 ：<code>.help</code><br>显示当前设置：<code>.show</code> 比如显示table时是否打印表头，各列之间以什么字符分隔，默认‘|’</p>
<h6 id="数据库和列表打开与查看">数据库和列表打开与查看</h6><p>打开一个数据库：<code>.open &lt;database_name&gt;</code> 通过这种方法打开的数据库，默认的名字是<em>main</em>，如果需要同时打开多个数据库，可以为每个数据库取一个别名，<code>ATTACH DATABASE &#39;/home/manager/Documents/sqlite3/ath.db&#39; as &#39;ath&#39;;</code><br>注意，<em>main</em>和<em>temp</em>是保留名，在每个数据库连接中都有，用于作为primary database，存放临时列表和其他数据对象。</p>
<p>显示当前已打开的全部数据库使用<code>.database</code>，同样地，如果需要查看当前session全部的table，可使用<code>.table(s)</code>。如果有多个数据库，注意看到拥有<em>alias name</em>的数据库中的列表，会以<strong>databasename.tablename</strong>的格式显示，但是在引用他们的名字时，使用全名（databasename.tablename）和列表名均可。</p>
<p>查看一个表共有哪些列，<code>.schema Organisms</code> 或<code>PRAGMA table_info(Organisms);</code><br><strong>注意</strong>，sqlite中<em>commands</em>全部是小写字母组成，且以点号’.’开头，而<em>statements</em>是大写字母组成，以分号结尾’;’ 你可能已经看出来了，sqlite是区分大小写的，这点和MySQL不同（关于sqlite、MySQL和NoSQL三者的比较，<a href="https://www.digitalocean.com/community/tutorials/sqlite-vs-mysql-vs-postgresql-a-comparison-of-relational-database-management-systems" target="_blank" rel="external">这里</a>的一篇文章很值得一看）。注释分为两种，<code>--</code>为行内注释，有限范围从字符往后到行尾，<code>/*</code>是C语言风格的注释，为块注释，位于<code>/*</code>和<code>*/</code>之间的内容会被sqlite引擎略过。</p>
<h6 id="常用的操作：简单统计、查询">常用的操作：简单统计、查询</h6><p>类似于linux的<code>head</code>,有的时候需要从列表中选出几条记录看一下，可以这样<code>SELECT * FROM Organisms LIMIT 3;</code> 返回如下：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">abbr        name&#10;----------  -----------------&#10;aaa         Acidovorax avenae&#10;aac         Alicyclobacillus&#10;aad         Alicyclobacillus</span><br></pre></td></tr></table></figure></p>
<p>此外，还有其他常见操作<br><figure class="highlight sql"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator"><span class="keyword">SELECT</span> <span class="keyword">COUNT</span>(abbr) <span class="keyword">FROM</span> Organisms;</span> <span class="comment">--统计表中全部记录的数目</span></span><br><span class="line"><span class="operator"><span class="keyword">SELECT</span> <span class="keyword">COUNT</span>(abbr) <span class="keyword">FROM</span> Organisms <span class="keyword">WHERE</span> abbr <span class="keyword">LIKE</span> <span class="string">'a__'</span>;</span> <span class="comment">--统计abbr以「a」开头，共3个字符长的记录数</span></span><br><span class="line">COUNT(abbr)</span><br><span class="line"><span class="comment">-----------</span></span><br><span class="line">232</span><br><span class="line"><span class="operator"><span class="keyword">SELECT</span> <span class="keyword">COUNT</span>(abbr) <span class="keyword">FROM</span> Organisms <span class="keyword">WHERE</span> abbr <span class="keyword">LIKE</span> <span class="string">'a%'</span>;</span> <span class="comment">--统计abbr以「a」开头的所有记录数</span></span><br><span class="line">COUNT(abbr)</span><br><span class="line"><span class="comment">-----------</span></span><br><span class="line">258</span><br></pre></td></tr></table></figure></p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/06/25/SGE新增用户配置/">SGE新增用户配置</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-06-25T08:20:47.000Z">Jun 25 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>日常的服务器管理中，经常需要添加或删除用户，对于linux用户的添加比较简单，使用<code>useradd</code>就可以：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#&#22686;&#21152;&#19968;&#20010;&#21517;&#20026;&#34;zhangshan&#34;&#30340;&#29992;&#25143;&#65292;&#24182;&#25226;&#20182;&#21152;&#20837;&#34;dev&#34;&#32676;&#32452;&#65292;&#21516;&#20107;&#21019;&#24314;&#20854;home&#30446;&#24405;(&#27880;&#65306; &#36825;&#37324;&#25152;&#26377;&#30340;&#25805;&#20316;&#65292;&#37117;&#26159;&#20197;root&#29992;&#25143;&#36523;&#20221;&#36827;&#34892;&#30340;)&#10;useradd -m -g dev zhangshan&#10;groupadd qa #&#26032;&#24314;&#19968;&#20010;&#32452;&#10;#&#26356;&#25913;&#23494;&#30721;&#10;passwd zhangshan&#10;&#9;&#60;p@ssw0rd&#62;</span><br></pre></td></tr></table></figure>
<p>添加完之后，新的用户就可以正常使用终端ssh登录集群（服务器）了，如果集群配置了SGE之类的作业调度系统，通常还需要对这些用户开通相应的集群队列权限，还是以用户”zhangshan”为例，将他加入”arusers”这个userset（SGE 称为Access List），这样他就可以使用”arusers”可调度的集群队列（all.q）了。</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#&#23558;&#34;zhangshan&#34; &#28155;&#21152;&#21040;&#34;arusers&#34;&#29992;&#25143;&#32452;&#10;[root@ngnode1 cuijie]# qconf -au zhangshan arusers&#10;added &#34;zhangshan&#34; to access list &#34;arusers&#34;&#10;#&#26174;&#31034;&#34;arusers&#34;&#29992;&#25143;&#21015;&#34920;&#65292;&#30475;&#21040;&#34;zhangshan&#34;&#24050;&#32463;&#21152;&#20837;&#36825;&#20010;&#21015;&#34920;&#10;[root@ngnode1 cuijie]# qconf -su arusers&#10;name    arusers&#10;type    ACL&#10;fshare  0&#10;oticket 0&#10;entries chengchaoze,cuijie,zhouyi,leijie,zhangshan&#10;#&#25226;&#34;zhangshan&#34;&#20174;&#34;arusers&#34;&#21024;&#38500;&#10;[root@ngnode1 cuijie]# qconf -du zhangshan arusers&#10;deleted user &#34;zhangshan&#34; from access list &#34;arusers&#34;</span><br></pre></td></tr></table></figure>
<p>更多选项和参数，可参考<code>qconf -help</code></p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/06/05/setup-and-run-PIPS-to-identify-Putative-Pathogenicity-Islands/">setup and run PIPS to identify Putative Pathogenicity Islands</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-06-04T22:16:21.000Z">Jun 5 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>Recently I need to do PAI search using <a href="http://www.genoma.ufpa.br/lgcm/pips/" target="_blank" rel="external">PIPS</a>, after going throught all the little traps, I think it worth to write the whole process down.</p>
<ol>
<li>follow the README instructions to install hammer and tRNA-scan. Maybe you want to change the default prefix of INSTALL derectory</li>
<li>Theoretically, it should be good enough to run, but you’ll definitely be stuck at some point. OK, here are the tricks to try below</li>
<li>replace <code>gbk2embl.pl</code> with a new one, such as this:</li>
</ol>
<pre><code><span class="comment">#!/usr/bin/env perl</span>
<span class="keyword">use</span> <span class="title">strict</span>;
<span class="keyword">use</span> <span class="title">warnings</span>;
<span class="keyword">use</span> <span class="title">Bio</span>::<span class="title">SeqIO</span>;

<span class="keyword">if</span> (@ARGV != <span class="number">2</span>) {    <span class="keyword">die</span> <span class="string">"USAGE: gb2embl.pl &lt;in.gbk&gt; &lt;out.embl&gt;\n"</span>; }

my <span class="variable">$seqio</span> = Bio::SeqIO-&gt;new(<span class="string">'-format'</span> =&gt; <span class="string">'genbank'</span>, <span class="string">'-file'</span> =&gt; <span class="string">"$ARGV[0]"</span>);
my <span class="variable">$seqout</span> = <span class="keyword">new</span> Bio::SeqIO(<span class="string">'-format'</span> =&gt; <span class="string">'embl'</span>, <span class="string">'-file'</span> =&gt; <span class="string">"&gt;$ARGV[1]"</span>);
<span class="keyword">while</span>( my <span class="variable">$seq</span> = <span class="variable">$seqio</span>-&gt;next_seq) {
  <span class="variable">$seqout</span>-&gt;write_seq(<span class="variable">$seq</span>)
}
</code></pre><p>For the last step, modify <code>mergfiles.pl</code>. change for e.g. <code>%hash2-&gt;{$chave}</code> to <code>$hash2{$chave}</code>, note that a couple of similar syntax mistake in this script.</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/05/29/Fast-genome-and-metagenome-distance-estimation-using-MinHash/">Fast genome and metagenome distance estimation using MinHash</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-05-29T06:47:55.000Z">May 29 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p><a href="http://mash.readthedocs.org" target="_blank" rel="external">mash</a> 是一个可快速计算两个（或多个）基因组之间相似度的软件，可输入基因组序列或测序结果（fastq文件），使用计算机科学中一个经典的方法MinHash，一般用来判断两个网页内容是否雷同，以决定是否从搜索结果中去掉某些类似的结果。</p>
<p>使用示例：</p>
<p><code>mash dist -p 4 -l 、/path/to/ref.fa query_fa.txt &gt; query_mashed.txt</code></p>
<p>用于计算所有query序列和ref序列的相似度，而相似度和物种间的亲缘关系是相关的，其中<query_fa.txt> 是一个记录了所有query序列的文件</query_fa.txt></p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2016/02/03/parsing-ABI-file/">parsing ABI file</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2016-02-03T14:45:00.000Z">Feb 3 2016</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>ABI测序仪下机的结果包含.ab1和.seq文件，通常可以导入BioEdit这样的图形化软件检查峰形和运行blast，而如果想要实现自动化的检测杂合位点并做比对，最终得到变异位点，就需要花些时间研究下。</p>
<p>主要的问题是，ab1文件是二进制的，关于文件格式的文档稀少，唯一靠谱的是<a href="http://www.appliedbiosystems.com/support/software_community/ABIF_File_Format.pdf" target="_blank" rel="external">ABI的官方文档</a> 仔细看过这个文档你会发现，看完之后作为新手的你尽管对于这个格式有所了解，知道它分为Header、Directory以及Data，以及有很多的tag。但知道这些对于想要从ab1提取trace value并进行heterogeneous base calling是远远不够的。所好的是，已经有人有过类似需求并作了尝试。</p>
<ul>
<li><p><a href="https://github.com/bow/abifpy" target="_blank" rel="external">abifpy</a> 实现的一个解析ab1文件的模块，现已并入BioPython。优点是兼容Python2和Python3，部分meta data已经给你提取出来了，而如果想要更多，可以使用get_data(‘<em>youkey</em>‘)获取。</p>
</li>
<li><p>如果只需要取出每个通道的trace value，有了官方文档和abifpy就够了，如果还想学习更多，可以读一读<a href="https://code.google.com/archive/p/seqtrace/" target="_blank" rel="external">SeqTrace</a>源码，从中可以学到怎样组织一个项目的代码，以及怎么写测试。</p>
</li>
<li><p>前面两个都研究一番之后，还是不能满足你的好奇心？可以把<a href="http://sourceforge.net/projects/staden/files/io_lib/" target="_blank" rel="external">io lib</a>这个库研究下，学习使用它的API.</p>
</li>
</ul>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2015/09/18/hello-again/">hello again</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2015-09-17T23:10:13.000Z">Sep 18 2015</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>翻了一下以前的文章发现，最近的上一篇文章是2013年11月的，那时候还在南京的学校，从找工作开始，然后面试，来北京，上班之后就更没有留出时间静下来写东西了。后来换了Mac，还特意把blog的资料作了备份，想着不久之后，重搭环境另起炉灶的，没想一下子拖到不久前才在新电脑上配好环境。</p>
<p>上班的这一年多，改变了很多。工作上从项目执行到流程负责人，再到信息分析主管而后技术副总监，见证了公司规模的扩张和变迁，也目睹了身边同事的来来去去，“送走”了三波运营岗同事，现在接触的大都是第四波的人。公司在成长，自己的工作也在经历变化，以前只需要自己干活，现在要考量人和事，把相应的事交给合适的人去做，然后跟踪进度，培养新人。</p>
<p>除了工作，生活上也有一些变化。刚到北京住的没有独立卫生间的小院，后来的院子有单独卫生间，尽管很少，还是方便多了。随公司搬到天津之后，住的更好。经过两段感情经历的洗礼，也成长了很多，对自己也有了更多地认识。写东西是一个特别好的总结的方式，通过它可以和自己对话，从而自省；和别人交流，从而结交同好。所以，以后我会坚持下去的。</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/11/24/retrieving-and-sending-infomation-over-the-web/">retrieving and sending infomation over the web</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-11-24T08:04:53.000Z">Nov 24 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>I found <a href="http://pythonforbiologists.com/index.php/applied-python-for-biologists/applied-python-2/" target="_blank" rel="external">this</a> when hanging around online. To be honest, I am dying to learn web crawling long time ago, can not wait to play with this.</p>
<p>modules involved:</p>
<ul>
<li>urllib2</li>
<li>re</li>
<li><a href="http://www.crummy.com/software/BeautifulSoup/" target="_blank" rel="external">BeautifulSoup</a> Need to install manually</li>
</ul>
<p>This post is very easy to follow with basic familarity of Python, the difficulty of web crawling is lots of, if not all, websites follow HTML specification exactly, there are often errors here and there if you try to parse useful bits of infomation out there, and here comes BeautifulSoup which handle most of these for us.</p>
<p>The second section discuss sending email with Python. One thing need to keep in mind is check your internet connection first if your code refuse to work :-)- I have complete the spec of No.5, and below are some commands I find useful:</p>
<ul>
<li><code>execfile(&#39;script.py&#39;)</code> #load script into console</li>
<li><code>python -i script.py</code> #execute script then come to Python console, ‘i’ short for ‘interactive’</li>
<li>sometimes you need to insert a time stamp into you file, and <code>time.ctime()</code> is probably the simplest solution<br>source code is <a href="https://github.com/popucui/RetrievandMail/blob/master/RetrievandMail.py" target="_blank" rel="external">here</a></li>
</ul>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/10/27/一道信息分析面试题的分析/">一道信息分析面试题的分析</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-10-27T06:47:39.000Z">Oct 27 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>前几天接到一个公司的电话面试，当天晚上HR发来一道题，限三日内完成并发回。我看了下，正好用到的知识自己都熟悉，花了几个钟头就给发回去了 :-)</p>
<p>原题大意这样 写一个程序，分别计算一个目录里所有fasta格式的核酸序列的GC含量，然后按照GC含量由低到高显示，并把结果写入一个文件<br>显示序列信息的格式为：<br>文件名  序列名  GC含量<br>hem.fasta hem1  24.3%</p>
<p>这道题挺简单的，细看之下主要有这么几点：</p>
<ul>
<li>计算序列GC含量，可以用一个函数实现</li>
<li>列出目录下的所有文件，然后判断他们是不是fasta格式文件。前者用os模块的<code>listdir</code>就可，后者就是要看文件的首行是不是形如“&gt;at1g23490”开头，即是否“&gt;”开头，后面紧跟字符，正则就行了。</li>
<li>结果排序。这个应该是这道简单笔试题最难的部分了。正好之前从一个<a href="http://wangwei007.blog.51cto.com/68019/1131610/" target="_blank" rel="external">博客</a>读到解决排序的办法，就依样做了。把每条序列的结果存到一个列表里，然后对列表排序。</li>
</ul>
<p>这是我的代码 <a href="https://github.com/popucui/PyManipulateFiles/blob/master/bioinfoInterviw.py" target="_blank" rel="external">信息分析面试</a></p>
<p> EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/10/11/linux常用命令/">linux常用命令</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-10-11T15:13:13.000Z">Oct 11 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <h3 id="git">git</h3><p>在github上参与别人的项目或则多人共同开发一个项目的话，fork一个项目的仓库过来，自己做完提交pull request之后，就等着作者review过之后进行merge了。但是之后作者再进行新的改动，我的分支上还是原样，那么怎样让把我的分支上的与最新的仓库一直呢?</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#assume you originally cloned from your forked repo,that remote will be called &#34;origin&#34;, &#10;#then you need to add firstguy repo as another remote&#10;git remote add firstguy git://github.com/firstguy/repo.git&#10;#After that is all set up, you should indeed be able to&#10;git pull firstguy mastergit push origin</span><br></pre></td></tr></table></figure>
<h3 id="ubuntu_apt">ubuntu apt</h3><p>apt search</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line"># package search&#10;apt-cache search &#10;apt-cache show #------(package &#33719;&#21462;&#21253;&#30340;&#30456;&#20851;&#20449;&#24687;&#65292;&#22914;&#35828;&#26126;&#12289;&#22823;&#23567;&#12289;&#29256;&#26412;&#31561;)</span><br></pre></td></tr></table></figure>
<p>apt install&amp;remove</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#package install&#10;sudo apt-get install &#10;sudo apt-get install # -----(package - - reinstall &#37325;&#26032;&#23433;&#35013;&#21253;)&#10;sudo apt-get -f install # -----(&#24378;&#21046;&#23433;&#35013;?#&#34;-f = --fix-missing&#34;&#24403;&#26159;&#20462;&#22797;&#23433;&#35013;&#21543;...)&#10;sudo apt-get remove #-----(package &#21024;&#38500;&#21253;)&#10;sudo apt-get remove - - purge # ------(package &#21024;&#38500;&#21253;&#65292;&#21253;&#25324;&#21024;&#38500;&#37197;&#32622;&#25991;&#20214;&#31561;)&#10;sudo apt-get autoremove --purge # ----(package &#21024;&#38500;&#21253;&#21450;&#20854;&#20381;&#36182;&#30340;&#36719;&#20214;&#21253;+&#37197;&#32622;&#25991;&#20214;&#31561;</span><br></pre></td></tr></table></figure>
<p>apt update and more</p>
<figure class="highlight"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">#update source&#10;sudo apt-get update &#10;sudo apt-get upgrade #------&#26356;&#26032;&#24050;&#23433;&#35013;&#30340;&#21253;&#10;sudo apt-get dist-upgrade # ---------&#21319;&#32423;&#31995;&#32479;&#10;sudo apt-get dselect-upgrade #------&#20351;&#29992; dselect &#21319;&#32423;&#10;apt-cache depends #-------(package &#20102;&#35299;&#20351;&#29992;&#20381;&#36182;)&#10;apt-cache rdepends # ------(package &#20102;&#35299;&#26576;&#20010;&#20855;&#20307;&#30340;&#20381;&#36182;?#&#24403;&#26159;&#26597;&#30475;&#35813;&#21253;&#34987;&#21738;&#20123;&#21253;&#20381;&#36182;&#21543;...)&#10;sudo apt-get build-dep # ------(package &#23433;&#35013;&#30456;&#20851;&#30340;&#32534;&#35793;&#29615;&#22659;)&#10;apt-get source #------(package &#19979;&#36733;&#35813;&#21253;&#30340;&#28304;&#20195;&#30721;)&#10;sudo apt-get clean &#38;&#38; sudo apt-get autoclean # --------&#28165;&#29702;&#19979;&#36733;&#25991;&#20214;&#30340;&#23384;&#26723; &#38;&#38; &#21482;&#28165;&#29702;&#36807;&#26102;&#30340;&#21253;&#10;sudo apt-get check #-------&#26816;&#26597;&#26159;&#21542;&#26377;&#25439;&#22351;&#30340;&#20381;&#36182;</span><br></pre></td></tr></table></figure>
<p><strong>nohup</strong>: short for “no hang up”</p>
<ul>
<li>usage: nohup command &amp;</li>
<li>default output: nohup.out which contains std out and std error, I refquently use this: <code>nohup sslocal &gt; log &amp;</code></li>
</ul>
<p>EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  

  <nav id="pagination">
  
  
    <a href="/archives/page/2/" class="alignright next">Next</a>
  
  <div class="clearfix"></div>
</nav>
</div>
    <div class="clearfix"></div>
  </div>
  <footer id="footer" class="inner"><div class="alignleft">
  
  &copy; 2016 Jack
  
</div>
<div class="clearfix"></div></footer>
  <script src="http://cdn.bootcss.com/jquery.imagesloaded/2.1.0/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>




<link rel="stylesheet" href="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.css" media="screen" type="text/css">
<script src="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>


</body>
</html>