<!DOCTYPE HTML>
<html lang="zh-CN">
<head>
  <meta charset="utf-8">
  
  <title>generate venn diagram in R | popucui</title>
  <meta name="author" content="Jack Cui">
  
  <meta name="description" content="Bioinformatics | Computer">
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  <meta property="og:title" content="generate venn diagram in R"/>
  <meta property="og:site_name" content="popucui"/>

  
    <meta property="og:image" content=""/>
  

  <link rel="shortcut icon" href="/favicon.png">
  
  
<link rel="stylesheet" href="/css/style.css">

  <!--[if lt IE 9]><script src="https://cdn.jsdelivr.net/npm/html5shiv@3.7.3/dist/html5shiv.min.js"></script><![endif]-->
  

<meta name="generator" content="Hexo 5.3.0"></head>


<body>
  <header id="header" class="inner"><div class="alignleft">
  <h1><a href="/">popucui</a></h1>
  <h2><a href="/">Stuff Related to Bioinformatics Computer and More</a></h2>
</div>
<nav id="main-nav" class="alignright">
  <ul>
    
      <li><a href="/">Home</a></li>
    
      <li><a href="/archives">Archives</a></li>
    
      <li><a href="/about/me">About</a></li>
    
  </ul>
  <div class="clearfix"></div>
</nav>
<div class="clearfix"></div>
</header>
  <div id="content" class="inner">
    <div id="main-col" class="alignleft"><div id="wrapper"><article id="post-generate-venn-diagram-in-r" class="h-entry post" itemprop="blogPost" itemscope itemtype="https://schema.org/BlogPosting">
  
  <div class="post-content">
    <header>
      
        <div class="icon"></div>
        <time class="dt-published" datetime="2013-09-18T14:35:57.000Z"><a href="/2013/09/18/generate-venn-diagram-in-r/">2013-09-18</a></time>
      
      
  
    <h1 class="p-name title" itemprop="headline name">generate venn diagram in R</h1>
  

    </header>
    <div class="e-content entry" itemprop="articleBody">
      
        <p>Our DGE project report provide a differently expressing genes list which is infered using DEGseq. When I was trying to follow the analysis pipline, I found that DEGseq has a so simple document that I cannot even figure out how to apply the function in my work, then I turn to DESeq, which provide a very good document with explanation of options and typical cases. Now I have to DG(Different Gene)lists: one from NOvogene, another is mine(generated by DESeq). Some genes are the same in them, while some not. My goal is to identify the same and plot a venn diagram. Here is the outline:</p>
<ul>
<li>idnetify the same genes, calculate the number of same and different genes.</li>
<li>draw a venn diagram in R to display above result.</li>
</ul>
<p>DEGseq list has 2k+ genes, and my DESeq list has 1230 genes, obviously I need a quick script to compare each gene in DESeq and remember the number of genes in both.</p>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br><span class="line">41</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line"></span><br><span class="line"></span><br><span class="line">dfgnlist1 = sys.argv[<span class="number">1</span>]</span><br><span class="line"></span><br><span class="line">dfgnlist2 = sys.argv[<span class="number">2</span>]</span><br><span class="line"></span><br><span class="line"><span class="keyword">with</span> <span class="built_in">open</span>(dfgnlist2, <span class="string">&#x27;r&#x27;</span>) <span class="keyword">as</span> deg:</span><br><span class="line"></span><br><span class="line">  dfgn2 = deg.readlines()</span><br><span class="line"></span><br><span class="line"><span class="keyword">with</span> <span class="built_in">open</span>(dfgnlist1,<span class="string">&#x27;r&#x27;</span>) <span class="keyword">as</span> de:</span><br><span class="line"></span><br><span class="line">  dfgn1 =de.readlines()</span><br><span class="line">i = <span class="number">0</span></span><br><span class="line"></span><br><span class="line">j = <span class="number">0</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> g <span class="keyword">in</span> dfgn1:</span><br><span class="line"></span><br><span class="line">  <span class="keyword">if</span> g <span class="keyword">in</span> dfgn2:</span><br><span class="line"></span><br><span class="line">    <span class="keyword">with</span> <span class="built_in">open</span>(<span class="string">&quot;samegenes.txt&quot;</span>,<span class="string">&#x27;a&#x27;</span>) <span class="keyword">as</span> samegenes:</span><br><span class="line"></span><br><span class="line">      samegenes.write(g)</span><br><span class="line"></span><br><span class="line">      i += <span class="number">1</span></span><br><span class="line"></span><br><span class="line">  <span class="keyword">else</span>:</span><br><span class="line"></span><br><span class="line">    <span class="keyword">with</span> <span class="built_in">open</span>(<span class="string">&quot;dfgenes.txt&quot;</span>,<span class="string">&#x27;a&#x27;</span>) <span class="keyword">as</span> dfgenes:</span><br><span class="line"></span><br><span class="line">      dfgenes.write(g)</span><br><span class="line"></span><br><span class="line">      j += <span class="number">1</span></span><br><span class="line"><span class="built_in">print</span> <span class="string">&#x27;========&#x27;</span></span><br><span class="line"></span><br><span class="line"><span class="built_in">print</span> i, <span class="string">&#x27;same genes&#x27;</span></span><br><span class="line"></span><br><span class="line"><span class="built_in">print</span> j, <span class="string">&#x27;different genes&#x27;</span></span><br></pre></td></tr></table></figure>
<a id="more"></a>
<p>Now I the number, time to fire up R and begin ploting. There are at least two ways to plot venn diagram, let us begin with a simpler one:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">require(venneuler)</span><br><span class="line"></span><br><span class="line">v &lt;- venneuler(<span class="built_in">c</span>(DEGseq=<span class="number">2635</span>, DESeq=<span class="number">1230</span>, <span class="string">&quot;DEGseq&amp;DESeq&quot;</span>=<span class="number">967</span>))</span><br><span class="line"></span><br><span class="line">plot(v)</span><br><span class="line"></span><br><span class="line">dev.copy2pdf(file=<span class="string">&quot;secVenneuler.pdf&quot;</span>)</span><br></pre></td></tr></table></figure>
<p>Need no extra comments, just so intuitive. Cannot wait to dive in? ok, here it is:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line">require(VennDiagram)</span><br><span class="line"></span><br><span class="line">venn.plot &lt;- draw.pairwise.venn(</span><br><span class="line"></span><br><span class="line">  <span class="number">1230</span>,<span class="number">2635</span>,<span class="number">967</span>, <span class="built_in">c</span>(<span class="string">&quot;DESeq&quot;</span>, <span class="string">&quot;DEGseq&quot;</span>), <span class="comment">#first for DESeq, second for DEGseq DEGseq, and third is num that both have. </span></span><br><span class="line"><span class="comment">#The last vector specify circle lables.</span></span><br><span class="line"></span><br><span class="line">  fill = <span class="built_in">c</span>(<span class="string">&quot;green&quot;</span>,<span class="string">&#x27;red&#x27;</span>),</span><br><span class="line"></span><br><span class="line">  alpha = <span class="built_in">c</span>(<span class="number">0.5</span>,<span class="number">0.5</span>),</span><br><span class="line"></span><br><span class="line">   cex = <span class="number">2</span>, lty = <span class="number">2</span>) <span class="comment">#lty for circles line type</span></span><br><span class="line"></span><br><span class="line">dev.copy2pdf(file=<span class="string">&#x27;sec6VennDiagram.pdf&#x27;</span>)</span><br></pre></td></tr></table></figure>
<p>There are more options and more functions in VennDiagram, check out the help manual if you want.</p>

      
    </div>
    <footer>
      
        
        
  
  <div class="tags">
    <a href="/tags/R/">R</a>, <a href="/tags/venn/">venn</a>
  </div>

        
  <div class="addthis addthis_toolbox addthis_default_style">
    
      <a class="addthis_button_facebook_like" fb:like:layout="button_count"></a>
    
    
      <a class="addthis_button_tweet"></a>
    
    
      <a class="addthis_button_google_plusone" g:plusone:size="medium"></a>
    
    
      <a class="addthis_button_pinterest_pinit" pi:pinit:layout="horizontal"></a>
    
    <a class="addthis_counter addthis_pill_style"></a>
  </div>
  <script src="//s7.addthis.com/js/300/addthis_widget.js"></script>

      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>


<section id="comment">
  <h1 class="title">留言</h1>

  
      <div id="fb-root"></div>
<script>
  (function(d, s, id) {
    var js, fjs = d.getElementsByTagName(s)[0];
    if (d.getElementById(id)) return;
    js = d.createElement(s); js.id = id;
    js.src = "//connect.facebook.net/en_US/all.js#xfbml=1&appId=123456789012345";
    fjs.parentNode.insertBefore(js, fjs);
  }(document, 'script', 'facebook-jssdk'));
</script>

<div class="fb-comments" data-href="http://popucui.github.io/2013/09/18/generate-venn-diagram-in-r/index.html" data-num-posts="5" data-width="840" data-colorscheme="light"></div>
      
  
</section>

</div></div>
    <aside id="sidebar" class="alignright">
  <div class="search">
  <form action="//google.com/search" method="get" accept-charset="utf-8">
    <input type="search" name="q" results="0" placeholder="搜索">
    <input type="hidden" name="as_sitesearch" value="popucui.github.io">
  </form>
</div>


  
<div class="widget tag">
  <h3 class="title">分类</h3>
  <ul class="entry">
  
    <li><a href="/categories/KEGG/">KEGG</a><small>1</small></li>
  
    <li><a href="/categories/R/">R</a><small>1</small></li>
  
    <li><a href="/categories/R-books/">R, books</a><small>1</small></li>
  
    <li><a href="/categories/TED-tips/">TED, tips</a><small>1</small></li>
  
    <li><a href="/categories/apple-news/">apple news</a><small>1</small></li>
  
    <li><a href="/categories/python/babel/">babel</a><small>1</small></li>
  
    <li><a href="/categories/books/">books</a><small>1</small></li>
  
    <li><a href="/categories/people/books/">books</a><small>1</small></li>
  
    <li><a href="/categories/editer-vim/">editer, vim</a><small>1</small></li>
  
    <li><a href="/categories/python/babel/key-error/">key error</a><small>1</small></li>
  
    <li><a href="/categories/os/">os</a><small>1</small></li>
  
    <li><a href="/categories/people/">people</a><small>1</small></li>
  
    <li><a href="/categories/perl/">perl</a><small>1</small></li>
  
    <li><a href="/categories/python/">python</a><small>2</small></li>
  
    <li><a href="/categories/vim/">vim</a><small>1</small></li>
  
  </ul>
</div>


  
<div class="widget tag">
  <h3 class="title">标签</h3>
  <ul class="entry">
  
    <li><a href="/tags/HTSeq/">HTSeq</a><small>1</small></li>
  
    <li><a href="/tags/R/">R</a><small>1</small></li>
  
    <li><a href="/tags/SGE-HPC-linux/">SGE, HPC, linux</a><small>1</small></li>
  
    <li><a href="/tags/Sanger-sequencing/">Sanger sequencing</a><small>1</small></li>
  
    <li><a href="/tags/bioinfomatics/">bioinfomatics</a><small>1</small></li>
  
    <li><a href="/tags/bioinformatics/">bioinformatics</a><small>1</small></li>
  
    <li><a href="/tags/crawler/">crawler</a><small>1</small></li>
  
    <li><a href="/tags/database-ClinVar/">database,ClinVar</a><small>1</small></li>
  
    <li><a href="/tags/distance-software/">distance, software</a><small>1</small></li>
  
    <li><a href="/tags/git/">git</a><small>1</small></li>
  
    <li><a href="/tags/maching-learning/">maching learning</a><small>1</small></li>
  
    <li><a href="/tags/os/">os</a><small>1</small></li>
  
    <li><a href="/tags/paper-notes/">paper notes</a><small>1</small></li>
  
    <li><a href="/tags/pips-software/">pips, software</a><small>1</small></li>
  
    <li><a href="/tags/programming/">programming</a><small>1</small></li>
  
    <li><a href="/tags/python/">python</a><small>2</small></li>
  
    <li><a href="/tags/shell/">shell</a><small>1</small></li>
  
    <li><a href="/tags/sqlite/">sqlite</a><small>1</small></li>
  
    <li><a href="/tags/ubuntu/">ubuntu</a><small>2</small></li>
  
    <li><a href="/tags/venn/">venn</a><small>1</small></li>
  
    <li><a href="/tags/windows/">windows</a><small>1</small></li>
  
    <li><a href="/tags/%E8%AF%BB%E4%B9%A6%E7%AC%94%E8%AE%B0/">读书笔记</a><small>1</small></li>
  
  </ul>
</div>

</aside>
    <div class="clearfix"></div>
  </div>
  <footer id="footer" class="inner"><div class="alignleft">
  
  &copy; 2021 Jack Cui
  
</div>
<div class="clearfix"></div></footer>
  
<script src="/js/jquery-3.4.1.min.js"></script>


<script src="/js/jquery.imagesloaded.min.js"></script>


<script src="/js/gallery.js"></script>






<link rel="stylesheet" href="/fancybox/jquery.fancybox.css">


<script src="/fancybox/jquery.fancybox.pack.js"></script>

<script>
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>

</body>
</html>
