<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8">
  
  <title>Page 2 | Archives | popucui</title>
  <meta name="author" content="Jack">
  
  <meta name="description" content="Bioinformatics | Computer">
  
  
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">

  
  <meta property="og:site_name" content="popucui"/>

  
    <meta property="og:image" content="undefined"/>
  

  <link href="/favicon.png" rel="icon">
  <link rel="alternate" href="/atom.xml" title="popucui" type="application/atom+xml">
  <link rel="stylesheet" href="/css/style.css" media="screen" type="text/css">
  <!--[if lt IE 9]><script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script><![endif]-->
  <script src="http://cdn.staticfile.org/jquery-i18n/1.1.1/jquery.i18n.min.js"></script>
  
</head>

<body>
  <header id="header" class="inner"><div class="alignleft">
  <h1><a href="/">popucui</a></h1>
  <h2><a href="/">Stuff Related to Bioinformatics Computer and More</a></h2>
</div>
<nav id="main-nav">
  <ul>
    
      <li><a href="/">Home</a></li>
    
      <li><a href="/archives">Archives</a></li>
    
      <li><a href="/about">About</a></li>
    
      <li><a href="/links">Links</a></li>
    
      <li><a href="/books">Books</a></li>
    
  </ul>
  <div class="clearfix"></div>
</nav>
<div class="clearfix"></div>
<div class="alignleft" style="margin-top: 15px">


  <iframe src="http://ghbtns.com/github-btn.html?user=popucui&type=follow"
  allowtransparency="true" frameborder="0" scrolling="0" width="132" height="20"></iframe>

</div>
<div class="clearfix"></div>
</header>
  <div id="content" class="inner">
    <div id="wrapper">
<h2 class="archive-title">Archives</h2>


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/14/using-htseq/">using HTSeq</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-14T00:29:50.000Z">Sep 14 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>用tophat做完mapping之后，接下来就是count reads，<a href="http://www-huber.embl.de/users/anders/HTSeq/doc/overview.html" target="_blank" rel="external">HTSeq</a>的文档很详细，还有一个use case，除了这个case，其他的全部跟着走了一遍。接下来使用htseq-qa和htseq-count的时候，遇到了问题，记录在这里：</p>
<h3 id="htseq-qa">htseq-qa</h3><p>安装了HTSeq后，htseq-qa的使用也很简单：</p>
<ul>
<li><code>htseq-qa [options] read_file</code> #assumes htseq is in your PATH</li>
<li><code>python -m HTSeq.scripts.qa [options] read_file</code> #if htseq is not in your PATH</li>
</ul>
<p>检查clean data的时候很顺利，在read_file是Col_0的raw data时，遇到了问题，原因是raw data里面有的reads quality string信息不完整，尝试过删除该reads或者把reads quality数据补全，都不成。</p>
<h3 id="htseq-count">htseq-count</h3><p>htseq-count的使用与qa类似，如果htseq不再你的PATH，可以这样：<code>python -m HTSeq.scripts.count [options] &lt;sam_file&gt; &lt;gff_file&gt;</code>。我遇到的问题有两个：</p>
<ol>
<li>tophat输出的是BAM格式的结果，而htseq-count需要SAM格式的比对结果，首先要转换，这个用samtools就行了：<code>samtools view col.bam &gt;&gt; col.sam</code> samtools默认的输入就是BAM格式，这里用重定向将屏幕输出写入col.sam</li>
<li>然后运行htseq-count报错：”ValueError: The attribute string seems to contain mismatched quotes”.搜了下，报错的行里面’gene_name’id含有多余的分号，手工修改了几次，还是报同样的错。这时候需要写一个脚本了：</li>
</ol>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line"><span class="keyword">import</span> re</span><br><span class="line"><span class="keyword">import</span> sys</span><br><span class="line"></span><br><span class="line">input_file = sys.argv[<span class="number">1</span>]</span><br><span class="line">output_file = sys.argv[<span class="number">2</span>]</span><br><span class="line">j = <span class="number">0</span></span><br><span class="line"><span class="keyword">with</span> open(input_file, <span class="string">'r'</span>) <span class="keyword">as</span> f:</span><br><span class="line">  <span class="keyword">for</span> line <span class="keyword">in</span> f.readlines():</span><br><span class="line">    gene_name = re.search(<span class="string">r'gene_name\s(\S+)'</span>,line)</span><br><span class="line">    gene_id = gene_name.group(<span class="number">1</span>)</span><br><span class="line">    match = re.search(<span class="string">r'(\w+);(\w+)'</span>, gene_id)</span><br><span class="line">    <span class="keyword">if</span> match:</span><br><span class="line">      newid = <span class="string">''</span>.join([<span class="string">'"'</span>, match.group(<span class="number">1</span>), match.group(<span class="number">2</span>), <span class="string">'"'</span>, <span class="string">';'</span>])</span><br><span class="line">      newline = line.replace(gene_id, newid)</span><br><span class="line">    <span class="keyword">else</span>:</span><br><span class="line">      newline = line</span><br><span class="line"></span><br><span class="line">    <span class="keyword">with</span> open(output_file, <span class="string">'a'</span>) <span class="keyword">as</span> newf:</span><br><span class="line">      newf.write(newline)</span><br><span class="line">      j += <span class="number">1</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">print</span> <span class="string">'totaly'</span>, j, <span class="string">'number of lines modified!'</span></span><br></pre></td></tr></table></figure>
<p>得到正确格式的gtf之后，就可以count了：<code>python -m HTSeq.scripts.count -m intersection-nonempty -s no sam_file gtf_file &gt;&gt; samplecount</code> 需要注意的事这里’-s’选项默认是’yes’,所以如果样品建库不是连特异性的话，一定要加<code>-s no</code>,要不有一半的reads就浪费了。另外就是把count结果写入文件，以备之后分析。</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/10/hello-again-ubuntu/">hello again ubuntu</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-10T10:48:24.000Z">Sep 10 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>我接触的第一个linux发行版就是基于ubuntu的Macbuntu，那时候还是10.04的时代，后来自己不小心弄坏了MBR，连Windows也进不去，捉了不少难，好在从此学会PE重建主引导的方法。之后进转投Fedora，直到现在又用回ubuntu。</p>
<p>最主要的原因就是受够了Fedora那难看的字体，而从一开始ubuntu给我的印象就是漂亮，比windows漂亮，至于Mac吗，还没用过暂不置评。放弃Fedora还有一个原因是yum包管理也开始觉得不爽，每次我要search install软件，都要先更新下，前天安装HTSeq又遇到更新源频频出问题，慢的叫人抓狂。</p>
<p>转到ubuntu之后，顺便也把博客从<a href="http://octopress.org/" target="_blank" rel="external">octopress</a>迁移到<a href="http://zespia.tw/hexo/" target="_blank" rel="external">hexo</a>。octopress是基于ruby的，而hexo基于node，速度比前者明显快多了，随着博客文章越来越多，我越来越在意这种差别。最后，把用到过得几个命令记下来：</p>
<ul>
<li><code>source somefile</code> reload the file,but only affect current session, will lose when restart terminal</li>
<li><code>dpkg -i xxx.deb</code> install xxx soft which is in ubuntu deb format</li>
<li><code>nvm use version num</code> use the specified version of node <code>nvm ls</code> list all installed versions of node</li>
</ul>
<p>EOF</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/08/use-kobas-to-identify-enriched-pathways/">use KOBAS to identify enriched pathways</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-08T13:33:00.000Z">Sep 8 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>看着手里公司做好的KEGG富集列表，总想自己试试看自己能做怎么样。之前想着自己用python写个脚本来做，开始写才发现，这活不是一个几行脚本能搞定的，不过发现了keggrest package，一个实现了KEGG API的好东西，基本上get link list都有了。今天发现了KOBAS，才知道国人已经为傻瓜化KEGG注释搭建<a href="http://kobas.cbi.pku.edu.cn/home.do">web server</a>.我要做的就是把合适的基因号导入，然后run就行了。</p>
<h3 id="ID_prepare">ID prepare</h3><p>我的基因号现在形如：</p>
<ul>
<li>AT1G01170</li>
<li>AT1G01480</li>
<li>AT1G01490</li>
</ul>
<p>我要把它转为：</p>
<ul>
<li>ath:AT1G01170</li>
<li>ath:AT1G01480</li>
<li>ath:AT1G01490</li>
<li>ath:AT1G01610</li>
</ul>
      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/09/07/tophat-install-and-usage-note/">tophat install and usage note</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-09-07T04:15:00.000Z">Sep 7 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <h3 id="install_tophat">install tophat</h3><p>I got our RNA-seq all data a couple of days ago, after playing with python keggrest package grabing KEGG anotaion data some while, I decided to try installing tophat on my PC and run mapping with real project data.</p>
<p>Generally just follow the instructions on tophat’s <a href="http://tophat.cbcb.umd.edu/tutorial.shtml" target="_blank" rel="external">offical page</a> is enough. Some points are listed below:</p>
<ul>
<li><p>bowtie. Download and unpack the latest bowtie2, copy it to somewhere you wish, and most importanly add ‘bowtie2’ ‘bowtie2-build’ and ‘bowtie2-inspect’ to your $PATH like this: <code>export PATH=$PATH:/home/popcui/bio/bowtie*</code> A better way to do this is <code>echo</code> previous command to ‘~/.bashrc’</p>
</li>
<li><p>You can either download the precompiled binary release which is more convinent but mybe a little bit slow, or you build tophat from source. And before that, you need to install SAM tools</p>
</li>
<li><p>SAM tools. First download SAM tools, then <code>make</code> it(you may want to check the INSTALL file for more info) You can add <code>make razip</code> option <strong>note</strong>: do not forget to add copy SAM tools misc bcftools to a path you wish and add it to your $PATH.</p>
</li>
</ul>
<p>Now if everything is OK, download the test data and fire up tophat on it to make sure all work as expected.</p>
<h3 id="prepare_your_reference">prepare your reference</h3><p>If you have bowtie2 installed and want to use it with tophat2, you must create Bowtie2 indexes for your data using Bowtie2-build. For me, as I already have ref.gtf and ref.fa in hand, all I need to do is <code>bowtie2-build ./rna-data/ref.fa ./bt2/ref.fa</code> Reference genome .fa file comes first, and the indexes I want to build will be saved to ‘./bt2/ref.fa’</p>
<p>Now we have everything ready, let’s begin mapping right away.<code>tophat2 -p 4 -G ./rna-data/ref.gtf -o col_th ./bt2/ref.fa ./rna-data/Col_0.clean.fq</code> ‘-p’ specify the number of threads you want to use, since I have an i3-380 cpu which support virtual threading, I use 4 cpus.’-G’ stands for genes and transcripts anotation. ‘-o’ obviously imply the path you want to have output files saved, then comes reference genome file and input clean data.</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/08/21/how-to-get-help-on-python-console/">how to get help on python console</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-08-21T14:46:00.000Z">Aug 21 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <ul>
<li>help(<em>argument</em>) where the argument maybe the module, object, class, method name you are intrested, of course you may need first to import it if it is a module or an object that comes with a module.</li>
<li>dir(<em>argument</em>) similar with previous help() function, the difference is dir() display short message than help() does. Also, you can do this help(dir) to check out dir() usage.</li>
<li>the previous two functions applies to all platforms,now comes another tip which works on unix-like machine: pydoc -k keyword.</li>
</ul>
<p>And as always, you can google it, if those tricks don’t satisfy you.</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/06/13/apple-new-ad/">apple new ad</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-06-13T15:31:00.000Z">Jun 13 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <h2 id="apple_wwdc_2013_keynote">apple wwdc 2013 keynote</h2><h3 id="beginning_declaration:">beginning declaration:</h3><p>if everyone is busy making something, how can anyone perfect anything? We start to confuse convenience with joy,abundance with choice, designing something requires focus. The first thing we ask is: what do we want people to feel? <strong>Delight/surprise/love/connection</strong>, then we begin to craft around our intention, it takes time, there are a thousand nos for every yes. We simplify, we perfect, we start over, until every idea we touch enhances each life it touches, only then do we sign our work.</p>
<h3 id="closing_ad:">closing ad:</h3><p>This is it, this is what matters—-the experience of our product. How it make someone feel? Will it make life better? Does it deserve to exist? We spend a lot of time on the feel of great things, until every idea we touch enhances each life it touches. You may rarely look at it, but you always feel it. This is our signature, and it means everything.</p>
<p>这个广告给人感觉 Tim Cook掌舵之下的苹果不再那么激进，少了”think different”那样的凌厉，锐不可当的气势，‘You can quote them, disagree with them, glorify or vilify them. About the only thing you can’t do is ignore them’取而代之的是一种胜券在握的淡定从容。的确，成长为市值最高的科技的苹果也需要这种转变，过去是适应规则，而今是指定推广标准了。</p>
<p><a href="http://en.wikipedia.org/wiki/Think_Different" target="_blank" rel="external">think different ad</a></p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/05/29/more-vim/">more vim</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-05-28T23:21:00.000Z">May 29 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>如果觉得昨天的vim学完后，不过瘾，这里还有几个好玩儿的命令。</p>
<ul>
<li><code>:!ls</code> 不退出当前编辑页面情况下执行<code>ls</code>命令。这<code>ls</code>可换成其他任意一个shell命令，这个在调试程序的时候经常用到，比如我改了一个地方，接着想看看执行情况，就可以这样。</li>
<li>导航 <code>h</code> <code>l</code> <code>j</code> <code>k</code> 分别是左移动，右移动，下移动一行，上移动一行；<code>n + shift g</code>移动到第n行 <code>shift g</code>不加行号，默认移动到末行; <code>ctrl g</code>显示当前正在编辑的文件名，光标所在行号; <code>gg</code>移动到首行 <code>G</code>移动到末行。</li>
<li>编辑 <code>dd</code>删除当前行 <code>dd n</code>往下删除n行  <code>yy n</code>往下复制n行 <code>p</code>讲粘帖板内容插入到当前行下面，<code>o</code>当前行下面插入一个空行并进入编辑模式 <code>O</code>当前行的<strong>上面 </strong>插入一行并进入编辑模式。</li>
<li><code>:split txt2</code> 分出一个窗口并打开文件’txt2’，<code>ctrl w</code>可以在两个窗口间跳转。</li>
</ul>
<p>此外还有，单个字符替换，单个词的删除，以及从当前位置删除到行首或行尾等也经常用到。比较新的linux发行版会附带一个vim教程’vimtutor’,让你通过操作学习使用vim.</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/05/29/windowsxia-zen-yang-an-zhuang-rbao/">windows下怎样安装R包</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-05-28T23:00:00.000Z">May 29 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>不管是看书还是讲R的ppt提到安装R包，都很简单：<code>install.packages(&quot;package name&quot;)</code>。可是在windows下试过之后，就会知道，安装目录下的library不可写，R程序会提示你使用一个自定义的library,会让你选‘是’还是‘否’，于是选是，下载完包之后，用<code>lib(&quot;package name&quot;)</code>加载一下，发现找不到前面安装的包，反复试过都是这样。</p>
<p>搜索一下，发现我们需要人工建一个library,然后告诉R这个library的位置。<a href="http://www.divms.uiowa.edu/help/windows/RPackage.html" target="_blank" rel="external">这里</a>有操作说明。其实就是配置一个叫’R_LIBS’的环境变量。然后在安装，<code>lib(&quot;package name&quot;)</code>加载，都没问题了。</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/05/16/ted-talks-10-top-time-saving-tech-tips/">TED talks:10 top time-saving tech tips</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-05-16T07:52:00.000Z">May 16 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <p>今天看到David Pogue在TED关于节省时间技巧的演讲，转述到这里，顺便加上我自己平时使用的技巧。</p>
<h2 id="web">web</h2><ul>
<li>space =&gt; scroll down one page</li>
<li>shift + space =&gt; scroll up one page</li>
<li>ctrl + + =&gt; 放大显示文本</li>
<li>ctrl + - =&gt; 缩小显示文本</li>
</ul>
<h3 id="more_than_that">more than that</h3><p>you can use these with any web browser, if you use ‘vimperater’ or ‘vimium’, you can throw away mouse when surfing the internet. Or you can use shortcuts listed below if you don’t want to learn to use above extensions.<br>
      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  
    <article class="post">
  
    <div class="gallery">
  <div class="photoset">
    
      <img src="">
    
  </div>
  <div class="control">
    <div class="prev"></div>
    <div class="next"></div>
  </div>
</div>
  
  <div class="post-content">
    <header>
      
  
    <h1 class="title"><a href="/2013/05/14/r-in-action-note0/">&#39;R in action&#39; note0</a></h1>
  

      
        <p class="published">
          Published: <time datetime="2013-05-13T23:19:00.000Z">May 14 2013</time>
        </p>
      
    </header>
    <div class="entry">
      
        <h2 id="Getting_help">Getting help</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">help.start()  #fire up general help&#10;help(&#34;function name&#34;) #usage about a function&#10;help(package=&#34;package name&#34;) #list index of functions and &#10;                             #datasets in a package&#10;?function name # eq to help(&#34;function name&#34;)&#10;??string #search for &#39;string&#39; in docs</span><br></pre></td></tr></table></figure>
<h2 id="packages">packages</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">installed.packages() #list all installed packages&#10;install.packages() #install a package, if no package name given&#10;                   #you&#39;ll be prompted to select a mirror site and&#10;                   #then select a package in a list&#10;lib(package name) #load a custom package. If the package is installed by you&#10;                  #you need to load it tell R to load it first before using it</span><br></pre></td></tr></table></figure>
<h2 id="file_manipulation">file manipulation</h2><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">getwd() #get current working directory&#10;setwd() #set current working directory. Note the dir path should be given&#10;        #as &#34;F:/R/project/somewhere&#34; instead of &#10;        #using back slash(used for escaping)&#10;dir.create(&#34;path/dir name&#34;) #create a dir. But you may find no function&#10;                            #such as dir.remove() or dir.delete() here :-)</span><br></pre></td></tr></table></figure>
<p>At last, if you find some piece of code in this book not running in wondows version R, you should try it out in unix version before considering it a mistake.</p>

      
    </div>
    <footer>
      
        




      
      <div class="clearfix"></div>
    </footer>
  </div>
</article>
<!-- hackish -->


  

  <nav id="pagination">
  
    <a href="/archives/" class="alignleft prev">Prev</a>
  
  
    <a href="/archives/page/3/" class="alignright next">Next</a>
  
  <div class="clearfix"></div>
</nav>
</div>
    <div class="clearfix"></div>
  </div>
  <footer id="footer" class="inner"><div class="alignleft">
  
  &copy; 2016 Jack
  
</div>
<div class="clearfix"></div></footer>
  <script src="http://cdn.bootcss.com/jquery.imagesloaded/2.1.0/jquery.imagesloaded.min.js"></script>
<script src="/js/gallery.js"></script>




<link rel="stylesheet" href="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.css" media="screen" type="text/css">
<script src="http://cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.pack.js"></script>
<script type="text/javascript">
(function($){
  $('.fancybox').fancybox();
})(jQuery);
</script>


</body>
</html>